src/v/kafka/server/handlers/handler_probe.cc

```diff
@@ -111,6 +117,32 @@ void handler_probe::setup_metrics(
       });
 }
 
+// For public metrics we only expose a small subset of the metrics for produce
+// and fetch requests to keep the count of metrics low
+void handler_probe::setup_public_metrics(
+  ssx::metrics::metric_groups& metrics, api_key key) {
+    namespace sm = ss::metrics;
+
+    if (config::shard_local_cfg().disable_metrics()) {
+        return;
+    }
+
+    const char* handler_name = handler_for_key(key).value()->name();
+
+    std::vector<sm::label_instance> labels{sm::label("handler")(handler_name)};
+
+    metrics.add_group(
+      prometheus_sanitize::metrics_name("kafka_handler"),
+      {
+        sm::make_histogram(
+          "latency_microseconds",
```

Comment:
Shouldn't this be seconds?

Admittedly it's not very clear from the documentation but it appears to me that although both the internal and public histograms use `microseconds` as their `duration_t`, `public_histogram_logform` scales this by 1000000 (without actually checking what `duration_t` is: the scale factor should probably be something like std::chrono::seconds(duration_t(1)).count() or something) leading to buckets with units of "seconds".

I think this is because public metrics have a rule that all time units should be seconds.

Reply:
Yes you are right. Quite a bit confusing indeed. Fixed the naming.